frontend/src/pages/modelServing/customServingRuntimes/CustomServingRuntimeContext.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react';
+import { Outlet } from 'react-router-dom';
+import {
+  Bullseye,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStateIcon,
+  Spinner,
+  Title,
+} from '@patternfly/react-core';
+import { ExclamationCircleIcon } from '@patternfly/react-icons';
+import { TemplateKind } from '~/k8sTypes';
+import { DEFAULT_CONTEXT_DATA } from '~/utilities/const';
+import { ContextResourceData } from '~/types';
+import { useContextResourceData } from '~/utilities/useContextResourceData';
+import useTemplates from './useTemplates';
+
+type CustomServingRuntimeContextType = {
+  refreshData: () => void;
+  servingRuntimeTemplates: ContextResourceData<TemplateKind>;
+};
+
+export const CustomServingRuntimeContext = React.createContext<CustomServingRuntimeContextType>({
+  refreshData: () => undefined,
+  servingRuntimeTemplates: DEFAULT_CONTEXT_DATA,
+});
+
+const CustomServingRuntimeContextProvider: React.FC = () => {
+  const servingRuntimeTemplates = useContextResourceData<TemplateKind>(useTemplates('opendatahub')); // TODO: change to operator namespace
+
+  const servingRuntimeTeamplateRefresh = servingRuntimeTemplates.refresh;
+
+  const refreshData = React.useCallback(() => {
+    servingRuntimeTeamplateRefresh();
+  }, [servingRuntimeTeamplateRefresh]);
```

Comment:
This does nothing but strip the promise response -- is this intended to be expanded?

Reply:
ugh, yeah, but i did it late yesterday and I wasn't thinking clearly, we need to fetch the order of the templates, but that's handled by the dashboardconfig, so we don't need multiple calls for the context.